### PR TITLE
feat: inbox react hooks api reference

### PR DESCRIPTION
**Pull request by LetItRock**
The React Hooks API reference.

https://github.com/user-attachments/assets/98083eff-702a-40fa-bb81-6d8e60afdfae

https://github.com/user-attachments/assets/c0cfa306-57f2-4ceb-9186-a851bddda65a

https://github.com/user-attachments/assets/896fd5d4-0d0b-44f4-b38f-dcd87c57e939

https://github.com/user-attachments/assets/d9ee80b4-dfd9-4e7d-810a-096dbc7c9907

https://github.com/user-attachments/assets/57995e16-1803-4a12-a72a-4f71e50ec2da

https://github.com/user-attachments/assets/e6c0996f-7ead-49af-a259-905154b0da76

### CODE REVIEW

File: inbox/react/hooks/novu-provider.mdx [hunk 1]
```diff
@@ -0,0 +1,17 @@
+---
+title: 'NovuProvider'
+---
+
+import NovuProviderExample from '/snippets/inbox/react/novu-provider-example.mdx';
+import NovuParams from '/snippets/inbox/headless/api-reference/novu-params.mdx';
+
+The `NovuProvider` is the top-level component that provides the [Novu instance](/inbox/headless/api-reference#novu) to the rest of the hooks through the context.
+Usually it's placed somewhere at the root of your application, which allows the hooks to be accessible throughout the application.
```
Comment (desiprisg) [inline inbox/react/hooks/novu-provider.mdx:9]:
```suggestion
Usually, it's placed somewhere in the root of your application, which makes the hooks accessible throughout the application.
```

File: inbox/react/hooks/overview.mdx [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+Follow these steps to get started with building your custom inbox UI:
+
+<Steps>
+  <Step title="Install the @novu/react package">
+    ```bash 
+    npm install @novu/react 
+    ```
+  </Step>
+  <Step title="Implement the NovuProvider">
+    To implement the [NovuProvider](/inbox/react/hooks/novu-provider) component, you need to place it in your application's code at the tree level where you want the hooks to be accessible. Here's how you can do it:
```
Comment (desiprisg) [inline inbox/react/hooks/overview.mdx:25]: Shouldn't we place the `<NovuProvider />` somewhere in the example below?

File: inbox/react/hooks/use-notifications.mdx [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+</ParamField>
+
+### `useNotifications` returns
+
+<ParamField path="notifications" type="Notification[] | undefined">
+  An array of notifications fetched by the hook. If there are no notifications
+  fetched yet, it will be `undefined`.
+</ParamField>
+<ParamField path="isLoading" type="boolean">
+  A boolean value indicating whether the notifications are being fetched.
```
Comment (desiprisg) [inline inbox/react/hooks/use-notifications.mdx:51]:
```suggestion
  A boolean value indicating whether the first notifications are being fetched.
```

File: inbox/react/hooks/use-preferences.mdx [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+### `usePreferences` returns
+
+<ParamField path="preferences" type="Preference[] | undefined">
+  An array of subscriber preferences fetched by the hook. If there are no
+  preferences fetched yet, it will be `undefined`.
+</ParamField>
+<ParamField path="isLoading" type="boolean">
+  A boolean value indicating whether the preferences are being fetched.
+</ParamField>
+<ParamField path="isFetching" type="boolean">
```
Comment (desiprisg) [inline inbox/react/hooks/use-preferences.mdx:35]: I think this only applies for the refetch.

File: inbox/react/hooks/use-counts.mdx [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+</ParamField>
+
+### `useCounts` returns
+
+<ParamField path="counts" type="Count[] | undefined">
+  An array of notification counts fetched by the hook. If there are no
+  notification counts fetched yet, it will be `undefined`.
+</ParamField>
+<ParamField path="isLoading" type="boolean">
+  A boolean value indicating whether the notification counts are being fetched.
```
Comment (desiprisg) [inline inbox/react/hooks/use-counts.mdx:37]:
```suggestion
  A boolean value indicating whether the notification counts are being fetched for the first time.
```

File: inbox/react/hooks/overview.mdx (hunk 2 above)
Comment (LetItRock) [reply inbox/react/hooks/overview.mdx:25]: @desiprisg it's there ;) 

Screenshot 2024-09-18 at 11 22 24

**Review by SokratisVidros — APPROVED** [review]

**Pull request merged**